src/routerify.ts

```diff
@@ -0,0 +1,36 @@
+import xs from 'xstream';
+import {adapt} from '@cycle/run/lib/adapt';
+import {RouteMatcher} from './interfaces';
+import {RouterSource} from './RouterSource';
+import {Location, createPath} from 'history';
+import {HistoryInput,  GenericInput} from '@cycle/history';
+import {Stream} from 'xstream';
+
+export declare type HistoryAction = HistoryInput | GenericInput | string;
+export declare type RouterSink = Stream<HistoryAction>;
+
+/**
+ * Wraps main to provide an advanced interface over @cycle/history
+ * @public
+ * @method routerify
+ * @return {main} The augmented main function
+ */
+function routerify(main: (a: any) => any, routeMatcher: RouteMatcher, basename = '/', historyName = 'history', routerName = 'router') {
+    if (typeof main !== 'function') {
+        throw new Error('First argument to routerify must be a valid cycle app');
+    }
+    const createHref = (location: Location) => basename + createPath(location);
+    return function(sources: any): any {
```

Comment:
@jvanbruegge I find it a little confusing that there will be both `router` and `history` sources/sinks available within main, but it also makes some sense.  The `history` source is the raw stream and the `router` stream is the interpreted one.  Since both would be available, it's no longer necessary to pass the `history$` through on the `router` source, but removing that would break backwards compatibility so I guess it should remain.